packages/web-frontend/cypress/scripts/testE2e/Snapshots.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import fs from 'fs';
+import { isPlainObject, set } from 'lodash';
+import { extname } from 'path';
+
+import { compareAsc } from '@tupaia/utils';
+
+/**
+ * NodePath is an array of two elements:
+ * NodePath[0]: array of paths to reach a certain node in a tree
+ * NodePath[1]: value of the node
+ *
+ * @typedef {[string[], any]} NodePath
+ */
+
+const nodePathsToTree = nodePaths =>
+  nodePaths.reduce((agg, entry) => {
+    const [path, value] = entry;
+    set(agg, path, value);
+    return agg;
+  }, {});
+
+const isJs = path => extname(path) === '.js';
+
+export class Snapshots {
+  snapshotTree;
+
+  /**
+   * Method-specific, used during snapshot extraction
+   */
+  extractionConfig = {};
+
+  /**
+   * @param {Object<string, any>} snapshotTree
+   */
+  constructor(snapshotTree) {
+    this.snapshotTree = snapshotTree;
+  }
+
+  static import = path => {
+    const contents = fs.readFileSync(path, { encoding: 'utf-8' });
+    // Note: JS parsing is a bit naive here, ideally we should use a code parsing library
+    const snapshotJson = isJs(path) ? contents.replace('module.exports = ', '') : contents;
```

Comment:
I don't expect that using a code parser would be too complicated here, but I skipped it because of work load. Happy to have another look in the near future